### ISSUE TITLE
[Boards] Implement ability to show/hide ticket properties on Kanban and GitHub boards via board configurator

### ISSUE THREAD

**Issue by carina-akaia** [near/neardevhub-widgets#250]
## Problem
Currently, users who enable the GitHub or Kanban board do not have the ability to customize which properties to display on the cards.

## User Story

As a community admin, I want to be able to choose which fields to show/hide on the boards.

## Acceptance Criteria

- Admin can choose to show/hide the following fields:
- For Kanban boards:
  - Post author icon
  - Post type icon
  - Post tags
  - ~~Number of replies~~ | Note: 09/12 We've decided to remove this from the requirements due to [technical challenge](https://github.com/near/neardevhub-widgets/issues/250#issuecomment-1714901375).
  - Number of likes 
  - For sponsorship posts, give the ability to show:
    - Amount
    - Currency
    - Supervisor
  - For solution posts, give the ability to show:
    - Seeking funding (Yes)
    - Amount
    - Currency
    - Requested Sponsor
- For GitHub boards: 
  - Ticket author
  - Ticket type icon (PR vs Issue)
  - Ticket # (E.g. NEPs #480)
  - Ticket labels

```[tasklist]
## Tasks
- [x] Implement changes for Kanban post boards
- [x] Implement changes for GitHub kanban boards
- [ ] https://github.com/near/neardevhub-contract/issues/60
```

## Preview

- [Kanban post board on Mainnet](https://near.org/carina.akaia.near/widget/gigs-board.pages.community.board?handle=devhub-test&nearDevGovGigsContractAccountId=devgovgigs.near)
- [Github kanban board on Mainnet](https://near.org/carina.akaia.near/widget/gigs-board.pages.community.github?handle=devhub-test&nearDevGovGigsContractAccountId=devgovgigs.near)

**Comment by ori-near** [comment]
Hi @carina-akaia – I went ahead and updated the acceptance criteria for this ticket. Please let me know if you have any questions. BTW, I think we should make the title field required by default. Perhaps we show it visually similar to how GitHub does it:
![Screenshot 2023-09-05 at 3 33 39 PM](https://github.com/near/neardevhub-widgets/assets/110252255/b1150b8d-b2a4-4484-8c48-9d19bb324bc7)

**Comment by carina-akaia** [comment]
Hi @ori-near, the updated scope will take more time. Can we add 2 days to the estimate?

**Comment by ori-near** [comment]
Hi @carina-akaia – I don't see a size estimate on the ticket, can you please add the final one?

**Comment by carina-akaia** [comment]
@ori-near done

**Comment by carina-akaia** [comment]
> Post type icon
> Ticket type icon (PR vs Issue)

@ori-near does this mean we only should hide the icon symbol or the type label ( Idea / Solution / Pull request / Issue / etc. ) together with it?

**Comment by carina-akaia** [comment]
> Post type icon
> Ticket type icon (PR vs Issue)

@ori-near does this mean we only should hide the icon symbol or the type label ( Idea / Solution / Pull request / Issue / etc. ) together with it?

**Comment by ori-near** [comment]
> > Post type icon
> > Ticket type icon (PR vs Issue)
> 
> @ori-near does this mean we only should hide the icon symbol or the type label ( Idea / Solution / Pull request / Issue / etc. ) together with it?

Hey @carina-akaia let's just hide both and the menu can say Post Type (which will include icon and text).

**Comment by carina-akaia** [comment]
@ori-near I can't see post replies among the data fetched from the smart contract. It'll take some time to figure it out and probably will require contract side changes. Can we extract "number of replies" AC into a separate ticket and mark it as blocked by the current one?

**Comment by ori-near** [comment]
Hi @carina-akaia – no problem, we can remove the "number of replies" requirement for now.

**Comment by carina-akaia** [comment]
@ori-near maybe it makes sense to merge Amount and Currency into one parameter, like "Requested funds" / "Granted funds"?

**Comment by carina-akaia** [comment]
> I can't see post replies among the data fetched from the smart contract

@ori-near unfortunately, none of the solution post properties appear anywhere besides the post markdown body. Let's include them into the separate task's scope as well?

**Comment by ori-near** [comment]
> @ori-near maybe it makes sense to merge Amount and Currency into one parameter, like "Requested funds" / "Granted funds"?

Hi @carina-akaia – we can just call it Amount and include both of those fields.

---

> > I can't see post replies among the data fetched from the smart contract
> 
> @ori-near unfortunately, none of the solution post properties appear anywhere besides the post markdown body. Let's include them into the separate task's scope as well?

Can you clarify what you mean? Can you please create a separate ticket outlining the scope and link from here?

**Comment by carina-akaia** [comment]
@ori-near addressed in https://github.com/near/neardevhub-contract/issues/60

**Comment by ori-near** [comment]
@carina-akaia Thanks for the preview. Overall looks great. A few feedback suggestions:

1. **Kanban/GitHub:** Please remove any fields that the user cannot currently enable/disable (e.g. amount of replies, sponsorship request indicator, requested grant value, requested sponsor)
2. **Kanban/Github:** Can you please move the configure button and new column buttons to where they were before? Also, the new transparent hover save at the bottom of the screen hides some of the preview elements and feels inconsistent with our other UI. Since it was not part of this scope, I would ask that we treat any UX considerations in a separate ticket.
3. **Kanban:** Please change the following copy:
  - "Ticket features" to "Card fields"
  - "Author's avatar" to "Author"
  - "Amount of likes" to "Likes"
  - "Amount of replies" to "Replies"
  - "Approved grant value" to "Approved amount"
  - "Sponsorship supervisor" to "Supervisor"
4. **GitHub:** Please change the following copy:
  - "Ticket features" to "Card fields"
  - "Id" to "GitHub ID"
5. Please make the cancel / save the same colors as other common elements (e.g. see community config).

**Comment by carina-akaia** [comment]
@ori-near done, please check previews

**Comment by ori-near** [comment]
Looks great, thank you @carina-akaia!
